staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/upgrade.go

```diff
@@ -32,6 +36,30 @@ const HeaderSpdy31 = "SPDY/3.1"
 type responseUpgrader struct {
 }
 
+// connWrapper is used to wrap a hijacked connection and its bufio.Reader. All
+// calls will be handled directly by the underlying net.Conn with the exception
+// of Read and Close calls, which will consider data in the bufio.Reader. This
+// ensures that data already inside the used bufio.Reader instance is also
+// read.
+type connWrapper struct {
+	net.Conn
+	closed    int32
+	bufReader *bufio.Reader
+}
+
+func (w *connWrapper) Read(b []byte) (n int, err error) {
+	if atomic.LoadInt32(&w.closed) == 1 {
+		return 0, io.EOF
+	}
+	return w.bufReader.Read(b)
```

Comment:
@dsnet if we make it past the closed check on line 51, and before this line is executed, `Close()` executes and closes the `net.Conn`, what will happen when we next execute `w.bufReader.Read(b)` - will it EOF?